src/base64.cr

```diff
@@ -95,11 +95,11 @@ module Base64
   # ```text
   # Tm93IGlzIHRoZSB0aW1lIGZvciBhbGwgZ29vZCBjb2RlcnMKdG8gbGVhcm4gQ3J5c3RhbA==
   # ```
-  def strict_encode(data) : String
+  def strict_encode(data : Slice(UInt8) | StaticArray(UInt8, 5) | String | StaticArray(UInt8, 20) | StaticArray(UInt8, 16)) : String
     strict_encode data, CHARS_STD, pad: true
   end
 
-  private def strict_encode(data, alphabet, pad = false)
+  private def strict_encode(data : Slice(UInt8) | StaticArray(UInt8, 5) | String | StaticArray(UInt8, 20) | StaticArray(UInt8, 16), alphabet : String, pad : Bool = false) : String
```

Comment:
The static arrays are interesting: they outline that the method should really just take a slice, and there are opportunities for optimizations.

Even the string in the other methods are interesting. There could be a specific overload that would call the slice method.

Reply:
I'm not sure about runtime optimizations because private methods are typically inlined anyway, so the static array won't be copied around.
But for sure the compiler would have an easier time if it has to deal with only one method accepting `Bytes` instead of a multitude of different `StaticArray` instance types.